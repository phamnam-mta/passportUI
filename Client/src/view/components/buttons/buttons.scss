@import "~@fluentui/react/dist/sass/References";

.close-button {
    display: flex;
    color: $ms-color-gray130;
}
