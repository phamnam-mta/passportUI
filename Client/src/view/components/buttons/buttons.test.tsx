import * as React from "react";
import { shallow } from "enzyme";
import { CloseButton, DefaultIconButton, DrawRegionButton } from "./buttons";

describe("Buttons", () => {
    let baseProps;
    beforeEach(() => {
        baseProps = {
            onClick: jest.fn(),
            disabled: false,
        };
    });

    describe("Close Button", () => {
        it("should renders correctly", () => {
            const wrapper = shallow(<CloseButton {...baseProps} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("DefaultIconButton Button", () => {
        it("should match snapshot, when default icon button is enabled", () => {
            const wrapper = shallow(<DefaultIconButton name="Add" {...baseProps} title={"title"} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when default icon button is disabled", () => {
            const wrapper = shallow(<DefaultIconButton name="Add" {...baseProps} disabled={true} title={"title"} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("DrawRegionButton Button", () => {
        it("should match snapshot, when draw region button is enabled but not checked", () => {
            const wrapper = shallow(<DrawRegionButton {...baseProps} checked={false} />);
            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when draw region button is enabled and checked", () => {
            const wrapper = shallow(<DrawRegionButton {...baseProps} checked={true} />);
            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when draw region button is disabled", () => {
            const wrapper = shallow(<DrawRegionButton {...baseProps} disabled={true} checked={false} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
