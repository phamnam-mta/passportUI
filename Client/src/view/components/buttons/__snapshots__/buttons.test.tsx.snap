// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Close Button should renders correctly 1`] = `
<CustomizedIconButton
  ariaLabel="Close"
  className="close-button"
  iconProps={
    Object {
      "iconName": "Cancel",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "rootHovered": Object {
        "color": "#605e5c",
      },
      "rootPressed": Object {
        "color": "#605e5c",
      },
    }
  }
/>
`;

exports[`Buttons DefaultIconButton Button should match snapshot, when default icon button is disabled 1`] = `
<CustomizedIconButton
  disabled={true}
  iconProps={
    Object {
      "iconName": "Add",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "root": Object {
        "color": "#323130",
      },
      "rootDisabled": Object {
        "backgroundColor": "transparent",
        "color": "#c8c6c4",
      },
      "rootHovered": Object {
        "color": "#323130",
      },
      "rootPressed": Object {
        "color": "#323130",
      },
    }
  }
  title="title"
/>
`;

exports[`Buttons DefaultIconButton Button should match snapshot, when default icon button is enabled 1`] = `
<CustomizedIconButton
  disabled={false}
  iconProps={
    Object {
      "iconName": "Add",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "root": Object {
        "color": "#323130",
      },
      "rootDisabled": Object {
        "backgroundColor": "transparent",
        "color": "#c8c6c4",
      },
      "rootHovered": Object {
        "color": "#323130",
      },
      "rootPressed": Object {
        "color": "#323130",
      },
    }
  }
  title="title"
/>
`;

exports[`Buttons DrawRegionButton Button should match snapshot, when draw region button is disabled 1`] = `
<CustomizedDefaultButton
  ariaLabel="Draw Region"
  checked={false}
  disabled={true}
  iconProps={
    Object {
      "iconName": "SingleColumnEdit",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "root": Object {
        "border": 0,
      },
    }
  }
  toggle={true}
>
  Region
</CustomizedDefaultButton>
`;

exports[`Buttons DrawRegionButton Button should match snapshot, when draw region button is enabled and checked 1`] = `
<CustomizedDefaultButton
  ariaLabel="Draw Region"
  checked={true}
  disabled={false}
  iconProps={
    Object {
      "iconName": "SingleColumnEdit",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "root": Object {
        "border": 0,
      },
    }
  }
  toggle={true}
>
  Region
</CustomizedDefaultButton>
`;

exports[`Buttons DrawRegionButton Button should match snapshot, when draw region button is enabled but not checked 1`] = `
<CustomizedDefaultButton
  ariaLabel="Draw Region"
  checked={false}
  disabled={false}
  iconProps={
    Object {
      "iconName": "SingleColumnEdit",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "root": Object {
        "border": 0,
      },
    }
  }
  toggle={true}
>
  Region
</CustomizedDefaultButton>
`;
